Purple Cobra Pull

deanthedream opened the pull request:
This merge contains:
1. New / updated plotting utilities
2. the switch from valfZmin to fZQuads, a list which provides detailed descriptions whether a local zodiacal light is a local minimum, minimum on the edge of keepout, and zodiacal light adjacent to solar keepout. The extractfZmin_fZQUADS function extracts valfZmin as originally done (this ICD change in this branch passes unittests across all current branches)
3. Caching omitted critical scriptfile components. SS.cachename contains more elements.

dsavransky commented:
@deanthedream  Is this ready to go, or are you still working on another fix?  Also, SLSQPScheduler needs some cleanup - please remove the pdb import, all print calls should be via self.vprint, and re-comment out the print on line 257.

deanthedream commented:
Dmitry, This is ready to go - pending a clean travis run of 949b290.
I addressed your comments replacing vestigal print statements with raise Exception clauses.
I skimmed through all other comments removed a few.
I would like to leave the remaining comments as side notes for future work and understanding code functionality.

deanthedream commented:
@dsavransky can you review these two small conflicts and merge? I am not sure which implementation is more correct.
